packages/wallets/brave/src/adapter.ts

```diff
@@ -0,0 +1,251 @@
+                try {
+                    await wallet.connect();
+                } catch (error: any) {
+                    throw new WalletConnectionError(error?.message, error);
+                }
+            }
+
+            if (!wallet.publicKey) throw new WalletAccountError();
+
+            let publicKey: PublicKey;
+            try {
+                publicKey = new PublicKey(wallet.publicKey.toBytes());
+            } catch (error: any) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+
+            wallet.on('disconnect', this._disconnected);
+
+            this._wallet = wallet;
+            this._publicKey = publicKey;
+
+            this.emit('connect', publicKey);
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+    }
+
+    async disconnect(): Promise<void> {
+        const wallet = this._wallet;
+        if (wallet) {
+            wallet.off('disconnect', this._disconnected);
+
+            this._wallet = null;
+            this._publicKey = null;
+
+            try {
+                await wallet.disconnect();
+            } catch (error: any) {
+                this.emit('error', new WalletDisconnectionError(error?.message, error));
+            }
+        }
+
+        this.emit('disconnect');
+    }
+
+    async sendTransaction(
+        transaction: Transaction,
+        connection: Connection,
+        options?: SendTransactionOptions
+    ): Promise<TransactionSignature> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                // transaction.serializeMessage() requires feePayer and
+                // recentBlockhash to be set.
+                // Brave will set both here when caller doesn't set any of them.
+                // Otherwise, request would be rejected.
+                transaction.feePayer = transaction.feePayer || this.publicKey || undefined
+                transaction.recentBlockhash =
+                  transaction.recentBlockhash || (await connection.getRecentBlockhash('finalized')).blockhash;
+                const { signature } = await wallet.signAndSendTransaction(transaction, options);
+                return signature;
+            } catch (error: any) {
+                throw new WalletSendTransactionError(error?.message, error);
+            }
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+
+        return await super.sendTransaction(transaction, connection, options);
+    }
+
+    async signTransaction(transaction: Transaction): Promise<Transaction> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                return (await wallet.signTransaction(transaction)) || transaction;
```

Comment:
When would you want to return the original transaction? Is this in cases where it's _already_ signed?

Reply:
Cool! If Brave's `wallet.signTransaction` can never return `null`, you should uninherit this part :)

Reply:
addressed [here](https://github.com/solana-labs/wallet-adapter/pull/480)